### ISSUE TITLE
Version 2.7.0 azure.azcollection.azure_rm_networkinterface module BUG

### ISSUE THREAD

**Issue by rodrigosouzam** [ansible/ansible_collections_azure#1710]
##### SUMMARY
In version 2.7.0 of the azure.azcollection.azure_rm_networkinterface module, when declaring the ip_configurations dictionary, the application security group ID is not being set correctly. Instead, it is mistakenly assigning the subscription ID used for logging in with Ansible. 
Reverting to version 2.6.0 resolves the issue, as it works as expected.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_networkinterface 

##### ANSIBLE VERSION

```paste below
4.5.1
```

##### COLLECTION VERSION

```paste below
2.7.0
```

##### STEPS TO REPRODUCE

Define network_interface module and 

```yaml
    - name: Assign VM NIC to ASG
      azure.azcollection.azure_rm_networkinterface:
        resource_group: "{{ resource_group_name }}"
        subscription_id: "{{ AZURE_SUBSCRIPTION_ID }}"
        name: "{{ networkinterface_name }}"
        virtual_network: 
          name: "{{ virtualnetwork_name }}"
          resource_group: "{{ network_resource_group_name }}"
          subscription_id: "{{ AZURE_SUBSCRIPTION_ID }}"
        subnet_name: "{{ subnet_name }}"
        security_group: 
          name: "{{ securitygroup_name }}"
          resource_group: "{{ resource_group_name }}"
          subscription_id: "{{ AZURE_SUBSCRIPTION_ID }}"
        ip_configurations:
          - name: default
            primary: True
            application_security_groups: "{{ asg_id }}"
```

##### EXPECTED RESULTS

You expect that application_security_groups ID is the one being declared

##### ACTUAL RESULTS

Its getting the subscription id from the credentials I'm logging with ansible

```paste below

```

**Comment by Fred-sun** [comment]
@rodrigosouzam The application group ID that you should use is not under the same subscription ID as the current resource. This bug has been fixed by #1711. Can you try it on? Thank you!

**Comment by rodrigosouzam** [comment]
Hello this is continue happening, specifying the application security group id is not working as expected.

**Comment by Fred-sun** [comment]
@rodrigosouzam Sorry #1711 didn't fully fix the issue. This issue, which was fixed in #1720, was due to a keyword error when we fetched the 'subscription' when obtaining the lexicalized resource ID. Thank you!
